grammar.js

```diff
@@ -725,9 +731,9 @@ module.exports = grammar({
 
     // grammar.txt marks this non-optional and includes omitted_array_size_expression in
     // expression but we can't match empty rules.
-    array_rank_specifier: $ => seq('[', commaSep(optional($._expression)), ']'),
+    array_rank_specifier: $ => seq('[', commaSep(optional(field('expression', $._expression))), ']'),
 
-    nullable_type: $ => seq($._nullable_base_type, '?'),
+    nullable_type: $ => seq(field('element_type', $._nullable_base_type), '?'),
```

Comment:
`underlying_type`?

Reply:
Roslyn uses `ElementType` in `int?`.

Reply:
My preference would be to stick with `type`, in line with other places in the grammar.